crates/spk-schema/crates/ident/src/ident.rs

```diff
@@ -201,3 +203,16 @@ where
         deserializer.deserialize_str(IdentVisitor(PhantomData))
     }
 }
+
+/// Idents that can be turned into a [`VersionIdent`] can implement this trait.
+pub trait AsVersion {
+    /// Return a borrowed version of this ident converted into a
+    /// [`VersionIdent`].
+    fn as_version(&self) -> &VersionIdent;
+}
+
+impl<T> AsVersion for Ident<VersionIdent, T> {
+    fn as_version(&self) -> &VersionIdent {
```

Comment:
Any thoughts about this being `AsVersionIdent::as_version_ident`? I only worry that it sounds like it should be just the version as a result

Reply:
There was already `as_version` methods on the idents before this change, so I took the nuclear option in #1134.